buildkite/resource_pipeline.go

```diff
@@ -591,11 +615,297 @@ func (*pipelineResource) Schema(ctx context.Context, req resource.SchemaRequest,
 				},
 			},
 		},
+		Version: 1,
+	}
+}
+
+func (p *pipelineResource) UpgradeState(ctx context.Context) map[int64]resource.StateUpgrader {
```

Comment:
@james2791 heads up: lets not bother with this approach for removing `archive_on_delete`. Its been deprecated and moved to provider config and we are in v0.X territory.

So its fine for this to just go away completely and not need mapping over. Should simplify the logic here significantly

Reply:
Fair - it does significantly 😉